### PR TITLE
config: lower outgoing htlc broadcast delta

### PR DESCRIPTION
**Pull request by joostjager**
Builds on top of #2879.

We create outgoing htlcs and accept incoming htlcs within the go-to-chain window, risking triggering a force close. This PR updates the block deltas to make sure that we don't trigger a force close ourselves.

The main change is lowering the broadcast delta for outgoing htlcs. It can be lower than the delta for incoming htlcs, as we don't risk not being able to settle with the preimage if the tx(es) don't confirm in time.

The spec even suggests a delta of -1 for outgoing htlcs (https://github.com/lightningnetwork/lightning-rfc/blob/master/02-peer-protocol.md#cltv_expiry_delta-selection). Meaning that we only go to chain in the block after the block where the htlc expires.

This PR doesn't lower it to -1, but uses 0 instead to prevent confusion around the negative number. This means that we go to chain when an outgoing htlc expires. 

`OutgoingCltvRejectDelta` is a few blocks more than `OutgoingBroadcastDelta`. This is to prevent us forwarding an htlc that would bring us inside the broadcast window. We would probably receive a settle or fail quickly, but if a new block arrives in the mean time, we'd force close the channel.

`FinalCltvRejectDelta` could have been equal to `IncomingBroadcastDelta`, because as an exit hop we can immediately decide what to do with the incoming htlc.

This PR does increase `FinalCltvRejectDelta` a bit because for hodl invoices we do not decide immediately. Eventually, for hodl invoices, there will be automatic cancellation to prevent channel closure. This automatic cancellation should trigger some blocks before we go to chain. We can probably use the same value `FinalCltvRejectDelta` and the delta at which we cancel back hodl invoice htlcs.

### CODE REVIEW

**Comment by joostjager** [comment]
Comments from @halseth 
- Add unit test to assert spec compliance of delta constant
- Reword comment on `IncomingBroadcastDelta` in `chain_arbitrator.go`

**Comment by joostjager** [comment]
Reminder: Also need to check the outgoing expiry when we are the sender of the payment. In that case, `HtlcSatisfyPolicy` isn't called.

**Comment by cfromknecht** [comment]
```
# github.com/lightningnetwork/lnd [github.com/lightningnetwork/lnd.test]
./lnd_test.go:9473:33: constant -4 overflows uint32
```

**Comment by joostjager** [comment]
@cfromknecht ptal

Implemented policy check for local htlcs.

Didn't implement unit test to assert spec compliance. It doesn't look very useful to me to just check that a constant has a value. Instead I added extensive comments to the config constants which should future devs make think about what these values mean.

File: htlcswitch/switch_test.go [hunk 1]
```diff
@@ -1382,8 +1397,9 @@ func TestSkipIneligibleLinksLocalForward(t *testing.T) {
 	chanID1, _, aliceChanID, _ := genIDs()
 
 	aliceChannelLink := newMockChannelLink(
-		s, chanID1, aliceChanID, alicePeer, false,
+		s, chanID1, aliceChanID, alicePeer, eligable,
```
Comment (cfromknecht) [inline htlcswitch/switch_test.go]: s/eligable/eligible

**Comment by joostjager** [comment]
@cfromknecht ptal

File: config.go [hunk 2]
```diff
@@ -68,11 +68,50 @@ const (
...
+	// are not in a hurry to force close, because there is nothing to claim
+	// for us. We do need to time the htlc out, because there may be an
+	// incoming htlc that will time out too (albeit later). Bolt #2 suggests
+	// a value of -1 here, but we allow one block less to prevent potential
+	// confusion around the negative value. It means we force close the
+	// channel at exactly the htlc expiry height.
+	defaultOutgoingBroadcastDelta = 0
+
+	// defaultOutgoingCltvRejectDelta defines the number of blocks before
+	// the expiry of an outgoing htlc at which we don't want to it offer to
```
Comment (halseth) [inline config.go]: s/it offer/offer it

File: htlcswitch/link.go [hunk 3]
```diff
@@ -2195,6 +2165,36 @@ func (l *channelLink) HtlcSatifiesPolicy(payHash [32]byte,
 		return &lnwire.FailExpiryTooFar{}
 	}
 
+	// Next, using the amount of the incoming HTLC, we'll calculate the
```
Comment (halseth) [inline htlcswitch/link.go]: commit message should explain the rationale behind this change.

File: config.go [hunk 4]
```diff
@@ -68,11 +68,50 @@ const (
 	defaultTorV2PrivateKeyFilename = "v2_onion_private_key"
 	defaultTorV3PrivateKeyFilename = "v3_onion_private_key"
 
-	defaultIncomingBroadcastDelta = 20
-	defaultFinalCltvRejectDelta   = 2
-
-	defaultOutgoingBroadcastDelta  = 10
-	defaultOutgoingCltvRejectDelta = 0
+	// defaultIncomingBroadcastDelta defines the number of blocks before the
```
Comment (halseth) [inline config.go]: excellent documentation 👍

File: htlcswitch/link.go (hunk 3 above)
Comment (joostjager) [reply htlcswitch/link.go]: added

**Comment by joostjager** [comment]
@halseth ptal

**Pull request merged**